stubs/app/client/ui/CreateTeam.ts

```diff
@@ -0,0 +1,351 @@
+import { bigBasicButton, bigPrimaryButton, bigPrimaryButtonLink } from 'app/client/ui2018/buttons';
+import { mediaSmall, theme, vars } from 'app/client/ui2018/cssVars';
+import { icon } from 'app/client/ui2018/icons';
+import { IModalControl, modal } from 'app/client/ui2018/modals';
+import { TEAM_PLAN } from 'app/common/Features';
+import { checkSubdomainValidity } from 'app/common/orgNameUtils';
+import { UserAPIImpl } from 'app/common/UserAPI';
+import {
+  Disposable, dom, DomContents, DomElementArg, IDisposableOwner, input, makeTestId,
+  Observable, styled
+} from 'grainjs';
+
+const testId = makeTestId('test-create-team-');
+
+export function buildNewSiteModal(context: Disposable, options: {
+  planName: string,
+  selectedPlan?: string,
+  onCreate?: () => void
+}) {
+  const { onCreate } = options;
+
+  return showModal(
+    context,
+    () => dom.create(NewSiteModalContent, onCreate),
+    dom.cls(cssModalIndex.className),
+  );
+}
+
+class NewSiteModalContent extends Disposable {
+  private _page = Observable.create(this, 'createTeam');
+  private _team = Observable.create(this, '');
+  private _domain = Observable.create(this, '');
+
+  constructor(
+    private _onCreate?: (planName: string) => void) {
+    super();
+  }
+
+  public buildDom() {
+    const team = this._team;
+    const domain = this._domain;
+    return dom.domComputed(this._page, pageValue => {
+
+      switch (pageValue) {
+        case 'createTeam': return buildTeamPage({
+          team,
+          domain,
+          create: () => this._createTeam()
+        });
+        case 'teamSuccess': return buildConfirm({ domain: domain.get() });
+      }
+    });
+  }
+
+  private async _createTeam() {
+    const api = new UserAPIImpl(getHomeUrl());
+    try {
+      await api.newOrg({ name: this._team.get(), domain: this._domain.get() });
+      this._page.set('teamSuccess');
+      if (this._onCreate) {
+        this._onCreate(TEAM_PLAN);
+      }
+    } catch (err) {
+      reportError(err as Error);
+    }
+  }
+}
+
+export function buildUpgradeModal(owner: Disposable, planName: string): void {
+  throw new UserError(`There is no plan logical in this instance of Grist`);
+}
+
+export function buildUpgradeButton(owner: IDisposableOwner, app: AppModel): UpgradeButton {
+  return {
+    showUpgradeCard: () => null,
+    showUpgradeButton: () => null,
+  };
+}
+
+export function buildConfirm({
+  domain,
+}: {
+  domain: string;
+}) {
+  return cssConfirmWrapper(
+    cssSparks(),
+    hspace('22px'),
+    cssHeaderLine('Team site created', testId("confirmation")),
+    hspace('40px'),
+    bigPrimaryButtonLink(
+      urlState().setLinkUrl({ org: domain || undefined }), 'Go to your site', testId("confirmation-link")
+    )
+  );
+}
+
+function buildTeamPage({
+  team,
+  domain,
+  create
+}: {
+  team: Observable<string>;
+  domain: Observable<string>;
+  create: () => any;
+}) {
+  const disabled = Observable.create(null, false);
```

Comment:
As it is not bound, it needs to be disposed.

Reply:
`disabled` it's used on the `click` function to prevent from several click

Reply:
What I mean is that, as it's an observable that is not bound to other object lifecycle, it may lead to some memory leakage, doesn't it?